database/meta.go

```diff
@@ -64,98 +63,61 @@ func NewMetaDB(config *config.Config, dbHandle *mgo.Session,
 	return metaDB
 }
 
-//EnsureRollingSettingsMatch ensures that the rolling database settings in the
-//metadb match the provided arguments if the database settings exist at all
-func (m *MetaDB) EnsureRollingSettingsMatch(db string, numchunks int) error {
-	m.lock.Lock()
-	defer m.lock.Unlock()
-	ssn := m.dbHandle.Copy()
-	defer ssn.Close()
-
+//GetRollingSettings gets the current rolling settings
+func (m *MetaDB) GetRollingSettings(db string) (exists bool, isRolling bool, currChunk int, totalChunks int, err error) {
```

Comment:
Since we're allow much more flexibility, this function simply returns the values from the database rather than do any logic like `EnsureRollingSettingsMatch` used to do.